lib/ieleTranslator.js

```diff
@@ -0,0 +1,368 @@
+// Runtime Verification, Inc.
+// IELE <=> Solidity encode/decode
+
+/**
+ * Convert string to hexstring
+ * @param {string} str
+ */
+function a2hex(str) {
+  var arr = [];
+  for (var i = 0, l = str.length; i < l; i ++) {
+    var hex = Number(str.charCodeAt(i)).toString(16);
+    arr.push(hex);
+  }
+  return arr.join('');
+}
+
+/**
+ * Convert hexstring to string
+ * @param {string} hexx 
+ */
+function hex2a(hexx) {
+  hexx = hexx.replace(/^0x/, '') // remove leading 0x
+  var hex = hexx.toString();//force conversion
+  var str = '';
+  for (var i = 0; i < hex.length; i += 2)
+      str += String.fromCharCode(parseInt(hex.substr(i, 2), 16));
+  return str;
+}
+
+/**
+ * Convert hex string to int (including negative number)
+ * @param {string} hex 
+ */
+function hexToInt(hex) {
+  hex = hex.replace(/^0x(00)*/, '')
+  if (!hex.length) {
+    hex = '00'
+  }
+  if (hex.length % 2 != 0) {
+      hex = "0" + hex;
+  }
+  var num = parseInt(hex, 16);
+  var maxVal = Math.pow(2, hex.length / 2 * 8);
+  if (num > maxVal / 2 - 1) {
+      num = num - maxVal
+  }
+  return num;
+}
+
+/**
+ * Encode a (Solidity) value to IELE value
+ * @param {string | string[]} value 
+ * @param {{type:string, components?:object[]}} type
+ * @return {string} encoded IELE value
+ */
+function encode(value, type) {
```

Comment:
:scream_cat:

Reply:
Wut? 😬

Reply:
It is stolen from https://github.com/runtimeverification/remix-ide (which I don't seem to have access to anymore :/ )

Reply:
This: https://github.com/input-output-hk/remix-ide/blob/phase/iele_testnet/remix/remix-lib/src/execution/ieleTranslator.js